src/EventStore.Core/Services/Storage/ReaderIndex/AllReader.cs

```diff
@@ -48,7 +48,7 @@ public class AllReader : IAllReader {
 		}
 
 		public IndexReadAllResult ReadAllEventsForward(TFPos pos, int maxCount) {
-			return ReadAllEventsForwardInternal(pos, maxCount, maxCount, EventFilter.None);
+			return ReadAllEventsForwardInternal(pos, maxCount, int.MaxValue, EventFilter.DefaultAllFilter);
```

Comment:
hmmm, i think that we need to make the change to int.MaxValue on `master` too, otherwise the read may return less than maxCount events

Reply:
and given that there are only very specific events filtered out, the read should not take that long with int.MaxValue, so it looks good :thinking:

Reply:
yep thats it exactly 👍 . indeed the forward version of this test `be_able_to_read_all_one_by_one_until_end_of_stream` fails without it - it gets immediate endofstream